src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,841 @@ static DamageTable D_80AD2550 = {
+Vec3f D_80AD2574 = { 0.0f, 0.0f, 0.0f };
+Vec3s D_80AD2580 = { 0x9920, -0x384, -0x320 };
+
+TexturePtr D_80AD2588 = object_osn_Tex_0166F8;
+TexturePtr D_80AD258C = object_osn_Tex_016EF8;
+TexturePtr D_80AD2590 = object_osn_Tex_0176F8;
+TexturePtr D_80AD2594 = object_osn_Tex_017EF8;
+TexturePtr D_80AD2598 = object_osn_Tex_0182F8;
+
+extern Gfx object_osn_DL_0192A0[];
+extern AnimationHeader object_osn_Anim_0201BC;
+extern FlexSkeletonHeader object_osn_Skel_0202F0;
+
+void func_80AD1634(EnOsn*, GlobalContext*);
+void func_80AD16A8(EnOsn*, GlobalContext*);
+void func_80AD14C8(EnOsn*, GlobalContext*);
+void func_80AD19A0(EnOsn* this, GlobalContext* globalCtx);
+
+void func_80AD0830(EnOsn* this, GlobalContext* globalCtx) {
+    this->collider.dim.pos.x = this->actor.world.pos.x;
+    this->collider.dim.pos.y = this->actor.world.pos.y;
+    this->collider.dim.pos.z = this->actor.world.pos.z;
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+s32 func_80AD08B0(GlobalContext* globalCtx) {
+    switch (Player_GetMask(globalCtx)) {
+        case PLAYER_MASK_GREAT_FAIRY:
+            return 0x1FD6;
+        case PLAYER_MASK_GIBDO:
+            return 0x1FD8;
+        case PLAYER_MASK_TRUTH:
+            return 0x1FDA;
+        case PLAYER_MASK_GIANT:
+            return 0x1FDC;
+        case PLAYER_MASK_KAFEIS_MASK:
+            return 0x1FDE;
+        case PLAYER_MASK_DON_GERO:
+            return 0x1FE0;
+        case PLAYER_MASK_BLAST:
+            return 0x1FE2;
+        case PLAYER_MASK_COUPLE:
+            return 0x1FE4;
+        case PLAYER_MASK_SCENTS:
+            return 0x1FE6;
+        case PLAYER_MASK_KAMARO:
+            return 0x1FE8;
+        case PLAYER_MASK_STONE:
+            return 0x1FEA;
+        case PLAYER_MASK_POSTMAN:
+            return 0x1FEC;
+        case PLAYER_MASK_BUNNY:
+            return 0x1FEE;
+        case PLAYER_MASK_CAPTAIN:
+            return 0x1FF0;
+        case PLAYER_MASK_BREMEN:
+            return 0x1FF2;
+        case PLAYER_MASK_CIRCUS_LEADER:
+            return 0x1FF4;
+        case PLAYER_MASK_KEATON:
+            return 0x1FF6;
+        case PLAYER_MASK_GARO:
+            return 0x1FF8;
+        case PLAYER_MASK_ALL_NIGHT:
+            return 0x1FFA;
+        case PLAYER_MASK_ROMANI:
+            return 0x1FFC;
+        default:
+            return 0;
+    }
+}
+
+void func_80AD0998(EnOsn* this) {
+    s16 curFrame = this->skelAnime.curFrame;
+    s16 lastFrame = Animation_GetLastFrame(sAnimations[this->unk_1EC].animation);
+
+    if (this->unk_1EC == 0x12 && curFrame == lastFrame) {
+        this->unk_1EC = 0x13;
+        Actor_ChangeAnimation(&this->skelAnime, sAnimations, 0x13);
+    }
+}
+
+void func_80AD0A24(EnOsn* this) {
+    s16 sp1E = this->skelAnime.curFrame;
+    s16 new_var = Animation_GetLastFrame(sAnimations[this->unk_1EC].animation);
```

Comment:
```suggestion
    s16 curFrame = this->skelAnime.curFrame;
    s16 lastFrame = Animation_GetLastFrame(sAnimations[this->unk_1EC].animation);
```
Just for consistency, you can name these temps just like the previous and next function